test/test_results.py

```diff
@@ -0,0 +1,158 @@
+import unittest
+import sys
+import os
+import subprocess
+import warnings
+
+# Add the parent directory to the Python path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from test.utils.custom_tsv_parser import ParseTSVFile
+
+PREF_CUTOFF = -4
+
+
+def get_scode_cscode_id(data):
+    # Create an id with scode and center with sorted values of scode and center
+    school_centers = []
+    for row in data:
+        scode = int(row["scode"])
+        center = int(row["cscode"])
+        sccode_center_id = sorted((scode, center))
+        sccode_center_id = "_".join(map(str, sccode_center_id))
+        school_centers.append(sccode_center_id)
+    return school_centers
+
+
+class TestSchoolCenter(unittest.TestCase):
+    """_Tests to validate the outcome of the output are matching
+    as per the requirements
+
+    Needs the ouptut from the results/school-center.tsv and
+    results/school-center-distance.tsv files to be present
+
+    """
+
+    def setUp(self):
+        self.school_center_file = "results/school-center.tsv"
+        self.school_center_distance_file = "results/school-center-distance.tsv"
+        self.school_center_pref_file = "sample_data/prefs.tsv"
+        schools_tsv = "sample_data/schools_grade12_2081.tsv"
+        centers_tsv = "sample_data/centers_grade12_2081.tsv"
+        cmd = f"python school_center.py {schools_tsv} {centers_tsv} {self.school_center_pref_file}"
+        subprocess.run(cmd, shell=True)
+
+    def tearDown(self):
+        os.remove(self.school_center_file)
+        os.remove(self.school_center_distance_file)
+
+    def test_results_exists(self):
+        """_Test if the application in running which output the results in the
```

Comment:
The test name should be descriptive enough so that one doesn't need to write any comments, if one needs to write a comment for a function/test, then usually there is a smell; it might not be the case here, but I think the descriptive test would be better off even looking at the console.

Reply:
I am using the python auto-doc string so just writing the comment to stick to consistency makes it easier for autoamated documentations later on. I can remove it but basically its checking if the output file exisits or not :).